### ISSUE TITLE
`kustomize build` with remote resource may panic on slow networks

### ISSUE THREAD

**Issue by johejo** [kubernetes-sigs/kustomize#3362]
**Describe the bug**

At https://github.com/kubernetes-sigs/kustomize/commit/6bbc8295932081d50f26756649f99f1dc2738de8#diff-867c23b9f5a4ba53f94117fdfeb7c9899f849e5e231fee3d720fab5ceae43b9f
A timeout has been added to the getter.

But, due to improper handling of channels, writing to closed channels occurs at some point.
https://github.com/kubernetes-sigs/kustomize/blob/67e445c10b08781c612c42d706090e035203902a/api/loader/getter.go#L97-L111

If the call to `client.Get` exceeds 21 seconds (I don't know why this timeout value was chosen), the `defer close(ch)` closes the channel because the case for the timer finishes and returns.
After that, the background goroutine writes to the channel and panic occurs.

It does not occur every time.

**Files that can reproduce the issue**

**Expected output**

```
$ kustomize build "github.com/kubernetes-sigs/kustomize//examples/multibases"
piVersion: v1
kind: Pod
metadata:
  labels:
    app: myapp
  name: cluster-a-dev-myapp-pod
spec:
  containers:
  - image: nginx:1.7.9
    name: nginx
---
apiVersion: v1
kind: Pod
metadata:
  labels:
    app: myapp
  name: cluster-a-prod-myapp-pod
spec:
  containers:
  - image: nginx:1.7.9
    name: nginx
---
apiVersion: v1
kind: Pod
metadata:
  labels:
    app: myapp
  name: cluster-a-staging-myapp-pod
spec:
  containers:
  - image: nginx:1.7.9
    name: nginx

```

**Actual output**

```
$ kustomize build "github.com/kubernetes-sigs/kustomize//examples/multibases"
panic: send on closed channel

goroutine 9 [running]:
sigs.k8s.io/kustomize/api/loader.getRemoteTarget.func1(0xc00021e160, 0xc00021c610, 0xc000168070)
        /home/heijo/ghq/github.com/kubernetes-sigs/kustomize/api/loader/getter.go:104 +0x65
created by sigs.k8s.io/kustomize/api/loader.getRemoteTarget
        /home/heijo/ghq/github.com/kubernetes-sigs/kustomize/api/loader/getter.go:102 +0x446
exit status 2
```

**Kustomize version**

- 67e445c1
- kustomize/v3.9.0

**Platform**

LInux

**Additional context**

**Comment by johejo** [comment]
Although not directly related to this issue, the variable `err` may be assigned from multiple goroutines at the same times, so it seems to be a race condition.

**Comment by monopole** [comment]
Thanks for the issue report @johejo , you're correct on all points.

**Comment by monopole** [comment]
fixed in #3400
